File: src/main/java/com/charity2/entities/Aziz/entity/Training.java
package com.charity2.entities.Aziz.entity;

import com.charity2.entities.Aziz.entity.Categorie;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Training {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;

    @Enumerated(EnumType.STRING)
    private Categorie category;

    private String trainer;
    private String location;

    @Temporal(TemporalType.DATE)
    private Date startDate;

    @Temporal(TemporalType.DATE)
    private Date endDate;

    private int capacity;


    // Getters et Setters
}

File: src/app/services/models/training.ts
export interface User {
  id: number;
  username: string;
  email: string;
}

export interface Training {
  id?: number;
  title: string;
  description: string;
  category: string;
  location: string;
  deadline: string;
  type: string;
  postedBy: User;
}
